internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -220,6 +225,30 @@ func (r *AgentServiceConfigReconciler) Reconcile(origCtx context.Context, req ct
 		return ctrl.Result{}, nil
 	}
 
+	warnings, failures, err := r.validateStorage(ctx, log, instance)
+	if err != nil {
+		log.WithError(err).Error("Failed to validate storage configuration")
+		return ctrl.Result{Requeue: true}, err
+	}
+	for _, warning := range warnings {
+		r.Recorder.Event(instance, "Warning", aiv1beta1.ReasonStorageFailure, warning)
+	}
+	if len(failures) > 0 {
```

Comment:
else { clean the status condition? }

Reply:
I think it isn't strictly needed because when the validation finishes successfully the rest of the controller will continue working, and it will eventually set another value for this condition, either true or false, doesn't really matter; it will replace the current value. However it is probably clearer it is explicitly removed as you suggest. I will do so.

Reply:
Done. Also added a test to verify that the condition is updated when the storage configuration is fixed.